src/Carnac.Logic/KeysController.cs

```diff
@@ -20,8 +22,12 @@ public KeysController(ObservableCollection<Message> messages, IMessageProvider m
             this.messages = messages;
             this.messageProvider = messageProvider;
             this.concurrencyService = concurrencyService;
-        }
 
+            var settingsProvider = new SettingsProvider(new RoamingAppDataStorage("Carnac"));
```

Comment:
Do we have an existing `ISettingsProvider` we could ask for in the constructor, rather than recreating this inline?